frontend/src/pages/RegisterPage.vue

```diff
@@ -0,0 +1,221 @@
+          Already have an account? Sign In
+        </router-link>
+        <wof-button :size="1.5" class="register-page__submit-button" @click="createAccount">
+          Create Account
+        </wof-button>
+      </div>
+    </div>
+  </wof-card>
+</template>
+
+<script>
+import WofInput from "../components/WofInput.vue";
+
+export default {
+  name: "RegisterPage",
+  components: { WofInput },
+  data() {
+    return {
+      username: {
+        value: '',
+        errorMsg: ''
+      },
+      email: {
+        value: '',
+        errorMsg: ''
+      },
+      password: {
+        value: '',
+        errorMsg: ''
+      },
+      passwordConfirm: {
+        value: '',
+        errorMsg: ''
+      },
+      avatar: {
+        img: "",
+        errorMsg: "",
+      },
+    };
+  },
+  computed: {
+    avatarImage() {
+      if (this.avatar.img.length > 0) {
+        return this.avatar.img;
+      } else {
+        return require(`../assets/placeholder.png`);
+      }
+    }
+  },
+  methods: {
+    setUsername(newValue) {
+      this.username.value = newValue;
+    },
+    setEmail(newValue) {
+      this.email.value = newValue;
+      if(!this.email.value.includes('@') && this.email.value.length > 0) {
+        this.email.errorMsg = 'Email should contain "@"';
+      } else {
+        this.email.errorMsg = '';
+      }
+    },
+    setPassword(newValue) {
+      this.password.value = newValue;
+      if(this.passwordConfirm.value.length > 0 && this.passwordConfirm.value != this.password.value) {
+        this.passwordConfirm.errorMsg = 'Passwords do not match'
+      } else {
+        this.passwordConfirm.errorMsg = '';
+      }
+    },
+    setPasswordConfirm(newValue) {
+      this.passwordConfirm.value = newValue;
+      if(this.passwordConfirm.value != this.password.value) {
+        this.passwordConfirm.errorMsg = 'Passwords do not match'
+      } else {
+        this.passwordConfirm.errorMsg = '';
+      }
+    },
+    uploadAvatar() {
+      console.log("click on button");
+      this.$refs.fileInput.click();
+    },
+    changeAvatar(event) {
+      console.log("click on input");
+      const file = event.target.files[0];
+      this.avatar.img = URL.createObjectURL(file);
+    },
+    formValidation() {
+     if(this.username.errorMsg == '' && this.password.errorMsg == '' && this.passwordConfirm.errorMsg == '' && this.email.errorMsg == '') {
+        if(this.username.value.length > 0 && this.password.value.length > 0 && this.passwordConfirm.value.length > 0 && this.email.value.length > 0) {
+          return true;
+        }
+      }
+      return false;
+    },
+    createAccount() {
+      if(this.formValidation()) {
+        //This will be replaced with call to http
+        console.log("Username:"+this.username.value);
+        console.log("Email:"+this.email.value);
+        console.log("Password:"+this.password.value);
+        console.log("ConfirmPassword:"+this.passwordConfirm.value);
```

Comment:
Shall we replace this with the function? I think it's already merged, right?

Reply:
Not in this task. I would like to separate component task and http layer task. We will implement whole registration in #47